hapi-fees/src/main/java/com/hedera/services/usage/schedule/entities/ScheduleEntitySizes.java

```diff
@@ -0,0 +1,72 @@
+package com.hedera.services.usage.schedule.entities;
+
+/*
+ * ‌
+ * Hedera Services API Fees
+ * ​
+ * Copyright (C) 2018 - 2020 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hederahashgraph.api.proto.java.SignatureMap;
+import com.hederahashgraph.api.proto.java.SignaturePair;
+
+import static com.hederahashgraph.fee.FeeBuilder.BASIC_ENTITY_ID_SIZE;
+import static com.hederahashgraph.fee.FeeBuilder.BASIC_RICH_INSTANT_SIZE;
+import static com.hederahashgraph.fee.FeeBuilder.BOOL_SIZE;
+import static com.hederahashgraph.fee.FeeBuilder.KEY_SIZE;
+
+public enum ScheduleEntitySizes {
+	SCHEDULE_ENTITY_SIZES;
+
+	/* { deleted } */
+	static int NUM_FLAGS_IN_BASE_SCHEDULE_REPRESENTATION = 1;
+	/* { payer, schedulingAccount } */
+	static int NUM_ENTITY_ID_FIELDS_IN_BASE_SCHEDULE_REPRESENTATION = 2;
+	/* { schedulingTXValidStart } */
+	static int NUM_RICH_INSTANT_FIELDS_IN_BASE_SCHEDULE_REPRESENTATION = 1;
+
+	public int fixedBytesInScheduleRepr() {
+		return NUM_FLAGS_IN_BASE_SCHEDULE_REPRESENTATION * BOOL_SIZE
+				+ NUM_ENTITY_ID_FIELDS_IN_BASE_SCHEDULE_REPRESENTATION * BASIC_ENTITY_ID_SIZE
+				+ NUM_RICH_INSTANT_FIELDS_IN_BASE_SCHEDULE_REPRESENTATION * BASIC_RICH_INSTANT_SIZE;
+	}
+
+	public int bytesInBaseReprGiven(byte[] transactionBody) {
+		return fixedBytesInScheduleRepr() + transactionBody.length;
+	}
+
+	/**
+	 * Signature map is not stored in state, we only need it for bpt
+	 */
+	public int bptScheduleReprGiven(SignatureMap sigMap) {
+		var bytes = 0;
+		for (SignaturePair signaturePair : sigMap.getSigPairList()) {
+			bytes += signaturePair.getPubKeyPrefix().size() +
+					signaturePair.getEd25519().size() +
+					signaturePair.getContract().size() +
```

Comment:
We only support `Ed25519` signatures so we might delete the others for clarity.

Also note that a "contract key" can't actually sign anything; the purpose of [the `JContractIDKey`](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/legacy/core/jproto/JContractIDKey.java) was going to be roughly as follows:
- **If** a crypto account `0.0.X` has a 1/N threshold key, one of which is a `0.0.Y` contract key, **then** contract `0.0.Y` can send funds to account `0.0.X` in a `ContractCall` not signed by any of `0.0.X`'s keys, **even if** `0.0.X` has `receiverSigRequired=true`.

This is not implemented, however. 😁

Reply:
Yes, you are absolutely right. We only support `Ed25519`. Those calculations were added only I though that we would "want" to calculate the `bytes-per-transaction` properly. Someone might have populated those properties and we wouldn't calculate it in the `bytes-per-transaction`. 

Let me know what you think on that.

Reply:
Good point, heh...then we could also just use `sigMap.toByteArray().length` here I think, which is how the legacy library [calculates `SignatureMap` sizes](https://github.com/hashgraph/hedera-services/blob/master/hapi-proto/src/main/java/com/hederahashgraph/fee/FeeBuilder.java#L277).)

Reply:
You are right! We will update it in a future PR 👍